src/test/java/fi/aalto/cs/intellij/actions/InstallPluginsNotificationActionTest.java

```diff
@@ -0,0 +1,45 @@
+package fi.aalto.cs.intellij.actions;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.notification.Notification;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import fi.aalto.cs.intellij.PluginsTestHelper;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.Test;
+
+public class InstallPluginsNotificationActionTest extends PluginsTestHelper {
+
+  @Test(expected = RuntimeException.class)
+  public void testActionPerformed() {
+    //given
+    AtomicInteger numberOfCallsOfInstall = new AtomicInteger(0);
+    AtomicInteger numberOfCallsOfRestartProposal = new AtomicInteger(0);
+    List<IdeaPluginDescriptor> descriptorList = getDummyPluginsListOfTwo();
+
+    AnActionEvent anActionEvent = mock(AnActionEvent.class);
+    Notification notification = mock(Notification.class);
+
+    //when
+    InstallPluginsNotificationAction installPluginsNotificationAction =
+        new InstallPluginsNotificationAction(
+            descriptorList,
+            (descriptor) -> {
+              numberOfCallsOfInstall.getAndIncrement();
+            },
+            numberOfCallsOfRestartProposal::getAndIncrement);
```

Comment:
Could Mockito be used here too, (so we could avoid those AtomicIntegers which, yeah I know, my bad, are a bit awkward)? I don't know.